### PR TITLE
parrot: init at 1.6.0

### PR DESCRIPTION
**Pull request by Gerg-L**
## Description of changes
Added parrot a discord music bot written in rust

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Gerg-L** [comment]
I removed darwin from platforms because i do not have a darwin system to test with and the build is failing as is

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Result of `nixpkgs-review pr 250640` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>parrot</li>
  </ul>
</details>

---

Small nit, otherwise LGTM. Only tested up to the point where I would have to set up a new bot account for the API access to work, which I CBA to do:

```
$ env DISCORD_TOKEN="1" DISCORD_APP_ID="2" parrot
Fatality! Parrot crashed because: Http(UnsuccessfulRequest(ErrorResponse { status_code: 401, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("discord.com")), port: None, path: "/api/v10/users/@me", query: None, fragment: None }, error: DiscordJsonError { code: 0, message: "401: Unauthorized", errors: [] } }))
```

File: pkgs/applications/audio/parrot/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  postInstall = ''
+    wrapProgram $out/bin/parrot \
+      --prefix PATH : ${lib.makeBinPath [ ffmpeg yt-dlp ]}
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description =
+      "A hassle-free, highly performant, host-it-yourself Discord music bot ";
```
[the remark this answers is not in the document]
Comment (Gerg-L) [reply pkgs/applications/audio/parrot/default.nix]: Trailing space was a copy-paste error, reduced the description down

**Comment by Gerg-L** [comment]
Alright *should* work on darwin now, lets see what happens with borgo

File: pkgs/applications/audio/parrot/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "A hassle-free Discord music bot";
+    homepage = "https://github.com/aquelemiguel/parrot";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ gerg-l ];
+    platforms = lib.platforms.all;
```
Comment (OPNA2608) [inline pkgs/applications/audio/parrot/default.nix:48]: `buildRustPackage` sets `meta.platforms` to a list of targetable platforms as a default. Keeping `meta.platforms` unset here is preferred <sub>(from what I've seen)</sub>, as `lib.platforms.all` includes platforms Rust can not target:

https://github.com/NixOS/nixpkgs/blob/7da17f587156ce3612117a3f48fdf8d61e00e8d2/pkgs/build-support/rust/build-rust-package/default.nix#L159-L170

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/parrot/default.nix (hunk 2 above)
Comment (Gerg-L) [reply pkgs/applications/audio/parrot/default.nix:48]: TIL, thank you!

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
I'd suggest using [`meta = with lib;`](https://nixos.org/manual/nixpkgs/stable/#chap-meta) as the manual uses; also, missing `meta.platforms`

**Review by eclairevoyant — COMMENTED** [review]
never mind, i missed the earlier comment about platforms. please ignore.

**Review by eclairevoyant — APPROVED** [review]

**Review by OPNA2608 — APPROVED** [review]
Result of `nixpkgs-review pr 250640` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>parrot</li>
  </ul>
</details>

---

Result of `nixpkgs-review pr 250640` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>parrot</li>
  </ul>
</details>

---

Seems fine to me.

**Review by Lord-Valen — APPROVED** [review]
LGTM

**Pull request merged**